mesonbuild/interpreter/interpreter.py

```diff
@@ -678,42 +678,52 @@ def validate_arguments(self, args, argcount, arg_types):
                 if not isinstance(actual, wanted):
                     raise InvalidArguments('Incorrect argument type.')
 
-    @FeatureNewKwargs('run_command', '0.50.0', ['env'])
-    @FeatureNewKwargs('run_command', '0.47.0', ['check', 'capture'])
-    @permittedKwargs({'check', 'capture', 'env'})
-    def func_run_command(self, node, args, kwargs):
+    # Executables aren'ta actually accepted, but we allow them here to allow for
+    # better error messages when overriden
+    @typed_pos_args(
+        'run_command',
+        (build.Executable, ExternalProgram, compilers.Compiler, mesonlib.File, str),
+        varargs=(build.Executable, ExternalProgram, compilers.Compiler, mesonlib.File, str))
+    @typed_kwargs(
+        'run_command',
+        KwargInfo('check', (bool, NoneType), since='0.47.0'),
+        KwargInfo('capture', bool, default=True, since='0.47.0'),
+        ENV_KW.evolve(since='0.50.0'),
+    )
+    def func_run_command(self, node: mparser.FunctionNode,
+                         args: T.Tuple[T.Union[build.Executable, ExternalProgram, compilers.Compiler, mesonlib.File, str],
+                                               T.List[T.Union[build.Executable, ExternalProgram, compilers.Compiler, mesonlib.File, str]]],
+                        kwargs: 'kwargs.RunCommand') -> RunProcess:
         return self.run_command_impl(node, args, kwargs)
 
     def run_command_impl(self,
-                         node: mparser.BaseNode,
-                         args: T.Sequence[TYPE_nvar],
-                         kwargs: TYPE_nkwargs,
+                         node: mparser.FunctionNode,
+                         args: T.Tuple[T.Union[build.Executable, ExternalProgram, compilers.Compiler, mesonlib.File, str],
+                                               T.List[T.Union[build.Executable, ExternalProgram, compilers.Compiler, mesonlib.File, str]]],
+                         kwargs: 'kwargs.RunCommand',
                          in_builddir: bool = False) -> RunProcess:
-        if len(args) < 1:
-            raise InterpreterException('Not enough arguments')
-        cmd, *cargs = args
-        capture = kwargs.get('capture', True)
+        cmd, cargs = args
+        capture = kwargs['capture']
+        env = kwargs['env']
         srcdir = self.environment.get_source_dir()
         builddir = self.environment.get_build_dir()
 
-        check = kwargs.get('check')
+        check = kwargs['check']
         if check is None:
             mlog.warning(implicit_check_false_warning, once=True)
             check = False
 
-        if not isinstance(check, bool):
-            raise InterpreterException('Check must be boolean.')
-
-        env = self.unpack_env_kwarg(kwargs)
 
         m = 'must be a string, or the output of find_program(), files() '\
             'or configure_file(), or a compiler object; not {!r}'
+        overriden_msg = ('Program {!r} was overridden with the compiled '
```

Comment:
overridden has two "d"s in it, so this looks odd -- although of course it's an internal implementation detail and a perfectly valid variable reference string.

Reply:
I reworded this slightly, and used the `overrode` instead of `overridden`.

Reply:
in the prose message at least, it *needs* to be overridden.

Present: Program X is override  with the compiled"
Past: Program X is overrode with the compiled

Reply:
back to overridden it is.

Reply:
For the record, this seems to have been corrected in the wrong commit which makes it look a bit odd when following the commit difflog.

Reply:
err that is to say I mean these: https://github.com/mesonbuild/meson/pull/9515#pullrequestreview-796720479

```diff
-    def func_run_command(self, node, args, kwargs):
+    # Executables aren'ta actually accepted, but we allow them here to allow for
+    # better error messages when overriden
+    @typed_pos_args(
+        'run_command',
```

Followed by:
```diff
-    @permittedKwargs({'check', 'capture', 'env'})
-    # Executables aren'ta actually accepted, but we allow them here to allow for
-    # better error messages when overriden
+    # Executables aren't actually accepted, but we allow them here to allow for
+    # better error messages when overridden
```